cmd/ddev/cmd/hostname.go

```diff
@@ -52,29 +47,31 @@ to allow DDEV to modify your hosts file. If you are connected to the internet an
 		name, dockerIP := args[0], args[1]
 		var err error
 
+		hostnameInHostsFile, err := hostname.IsHostnameInHostsFile(name)
+		if err != nil {
+			util.Warning("Could not check existence of %s in hosts file: %v", name, err)
+		}
 		// If requested, remove the provided host name and exit
-		if removeHostnameFlag {
-			err = ddevapp.RemoveHostEntry(name, dockerIP)
+		if removeHostnameFlag && hostnameInHostsFile {
```

Comment:
Fix condition check:

```suggestion
		if removeHostnameFlag {
			if !hostnameInHostsFile {
				return
			}
```

Reply:
Oops. I had noticed that the combined condition wasn't going to work so was reverting but didn't do it quite right.